### PR TITLE
fix: add labels to certificate manager resources

### PR DESCRIPTION
**Pull request by m-ildefons**
Add missing labels to certificate management resources. This allows for clean upgrades and deinstallation when the chart has changed significantly or resources are moved between files between releases.

Signed-off-by: Moritz Röhrich <moritz.rohrich@suse.com>

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] CHANGELOG.md has been updated should there be relevant changes in this PR.

### CODE REVIEW

**Review by giubacc — APPROVED** [review]
I assume that this is safe and you have tested it, so LGTM (but please confirm this before merge)

**Comment by m-ildefons** [comment]
Yes. Labels are metadata that can be associated with _any_ resource. They're used to identify groups of resources that belong together.

**Pull request merged**
